### PR TITLE
[ci] upgrade CI to Ubuntu 18.04

### PR DESCRIPTION
**Pull request by jameslamb**
The scheduled CI job for this project failed like this tonight

image

image

All of the R jobs broke with

> dist: trusty is no longer supported for language: r

This looks like a permanent change in Travis R builds: https://twitter.com/opencpu/status/1299071086332653569.

This PR proposes updating all the way to Ubuntu 18.04 (we're currently on 14.04), so that hopefully it will be another couple years until we have to think of this again.

### CODE REVIEW

**Comment by jameslamb** [comment]
ok I think the version of Java we're getting is too new for the oldest Elasticsearch releases

image

image

```text
dpkg: warning: downgrading elasticsearch from 5.5.0 to 1.0.0
(Reading database ... 124198 files and directories currently installed.)
Preparing to unpack elasticsearch.deb ...
Unpacking elasticsearch (1.0.0) over (5.5.0) ...
dpkg: warning: unable to delete old directory '/var/lib/elasticsearch': Directory not empty
dpkg: warning: unable to delete old directory '/var/log/elasticsearch': Directory not empty
dpkg: warning: unable to delete old directory '/usr/lib/systemd/system': Directory not empty
Setting up elasticsearch (1.0.0) ...
Installing new version of config file /etc/init.d/elasticsearch ...
Installing new version of config file /etc/default/elasticsearch ...
Installing new version of config file /etc/elasticsearch/elasticsearch.yml ...
Processing triggers for systemd (237-3ubuntu10.41) ...
Job for elasticsearch.service failed because the control process exited with error code.
See "systemctl status elasticsearch.service" and "journalctl -xe" for details.
The command ".ci/seed_es_on_travis.sh" failed and exited with 1 during .
```

https://travis-ci.org/github/uptake/uptasticsearch/builds/724801618

**Comment by jameslamb** [comment]
ok I tried downgrading to Xenial (Ubuntu 16.04) and now only the Elasticsearch 1.0.0 tests are failing

https://travis-ci.com/github/jameslamb/uptasticsearch/jobs/382134542

next I'm trying to pin to to Java 8: https://github.com/uptake/uptasticsearch/pull/214/commits/c416c2746d0b349c3d38e5b45b5a06d576b316e0

**Comment by jameslamb** [comment]
I think we should just use docker to run Elasticsearch. It'll relieve a lot of the pain of managing this.

I'll try to merge `setup_local.sh` and `.ci/seed_es_on_travis.sh` for that purpose.

**Comment by austin3dickey** [comment]
yeah that sounds awesome

**Comment by jameslamb** [comment]
ok I got this working!

File: setup_local.sh [hunk 1]
```diff
@@ -94,4 +149,4 @@ curl -X GET "http://${ES_HOST}:9200/shakespeare/_search?size=1"
 cd ${WDIR}
 
 echo ""
-echo "Your local environment is ready!"
+echo "Your local environment is ready."
```
Comment (austin3dickey) [inline setup_local.sh:152]: :surprised-pikachu:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ The set of Elasticsearch versions this project tests against changes regularly a
 
 So, for example, as of August 2020 that meant we tested against:
```
Comment (austin3dickey) [inline CONTRIBUTING.md:92]: wanna change to September?

**Review by austin3dickey — APPROVED** [review]
This is wonderful, much simpler.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jameslamb) [reply CONTRIBUTING.md:92]: oh yep!

File: setup_local.sh (hunk 1 above)
Comment (jameslamb) [reply setup_local.sh:152]: the longer I worked on this, the more it felt like that `!` was taunting me

**Pull request merged**